cardano-testnet/src/Testnet/Util/Cli.hs

```diff
@@ -0,0 +1,186 @@
+module Testnet.Util.Cli
+  ( cliAddressKeyGen
+  , cliNodeKeyGen
+  , cliNodeKeyGenVrf
+  , cliNodeKeyGenKes
+  , cliStakeAddressKeyGen
+  , Comment (..)
+  , KeyNames (..)
+
+  , fakeItH
+  , fakeIt
+  , TmpDir
+  , TestnetMagic
+  , File (..)
+  , Address
+  , Operator -- == Cold Keys
+  , Kes
+  , StakeAddress
+  , Vrf
+
+  , VKey
+  , SKey
+
+  , OperatorCounter
+  , ByronKey
+  , ByronAddress
+
+  , ByronDelegationKey
+  , ByronDelegationCert
+
+  , getVKeyPath
+  , getSKeyPath
+
+  , makeFilePath
+  , cliKeyGen
+
+  , cliSigningKeyAddress
+  ) where
+
+import           Prelude
+
+import           Data.String
+import           Control.Monad
+import           System.FilePath.Posix
+
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.File as H (writeFile)
+import           Hedgehog
+import           Hedgehog.Gen
+
+import           Testnet.Util.Process
+
+-- TODO: check no whitespace , slash etc allowed
+-- TODO: use Comment instead of Filepath
+newtype Comment = Comment { unComment :: String}
+  deriving (Show, Eq, Ord )
```

Comment:
This will need to be MacOS and Windows friendly.  MacOS filenames are case insensitive.  Windows filenames are case-insensitive and many characters have special meaning.

Perhaps we can call it `Kebab` because it's using kebab-casing and we lock it down do only support lower case alpha-numerics.